### ISSUE TITLE
Problem with modules - missing library

### ISSUE THREAD

**Issue by beherit** [jabber-at/ejabberd#16]
> What version of ejabberd are you using?

ejabberd 17.01

> What operating system (version) are you using?

Debian 8.7

> How did you install ejabberd (source, package, distribution)?

Package from jabber.at

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

I reported issue with compiling modules in ejabberd 16.12 and never and I received answer that problem is with my ejabberd installation: https://github.com/processone/ejabberd-contrib/issues/197#issuecomment-285690029  I installed ejabberd from jabber.at repo and there is missed fast_xml library.

**Comment by mathiasertl** [comment]
The library should be installed with the `erlang-p1-xml` library. Is it installed on your system? `ejabberd` should depend on it.

Other then that, can you provide an error message?

**Comment by beherit** [comment]
`erlang-p1-xml` library is installed on my system. I don't see any error message - I have only errors when compiling modules.

**Comment by mathiasertl** [comment]
The issue seems to be that this patch: https://github.com/jabber-at/ejabberd/blob/master/debian/patches/src.ext_mod.erl.patch wasn't ported with the new package. 

The patch does not apply to 17.01. Unfortunately my Erlang foo isn't good enough to update the patch. If  you know how to patch it, I'm happy to accept a pull request. Otherwise I have asked for help upstream and will push an update once I have it.

**Comment by beherit** [comment]
Oh I don't understand Erlang :(

**Comment by mathiasertl** [comment]
No worries, I hope we'll get a patch soon! :-)

**Comment by beherit** [comment]
@mathiasertl any news? :) By the way, I have problem with update ejabberd 17.01 to 17.03:
`The following packages have unmet dependencies:
 ejabberd : Conflicts: erlang-p1-xmpp (> 1.1.10) but 1.1.16-0.3~afa80 is to be installed`

**Comment by github-tomster** [comment]
i second that.
the systems says:
```
Updatable package information
Package name 	ejabberd
Update system 	APT
Package description 	distributed, fault-tolerant Jabber/XMPP server
Current state 		New version 17.04-2~ubuntu16.04
Installed version 	1:17.01-2~ubuntu16.04
Available version 	1:17.04-2~ubuntu16.04
```
```
The following packages have unmet dependencies:
 ejabberd : Conflicts: erlang-p1-xmpp (> 1.1.10) but 1.1.16-0.3~ubuntu16.04 is to be installed
E: Unable to correct problems, you have held broken packages.
```
